poc/OwinTestApp/src/OwinClassLibrary/Utilities/MAuthClient.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Net.Http;
+using System.Threading.Tasks;
+using OwinClassLibrary.Authentication;
+using OwinClassLibrary.Cat;
+using OwinClassLibrary.Zipkin;
+
+namespace OwinClassLibrary.Utilities
+{
+    public class MAuthClient : IMAuthClient
+    {
+        private readonly HttpMessageHandler _handler;
+
+        public MAuthClient(ITraceProvider provider)
+        {
+            _handler = new MAuthMessageHandler(new ZipkinMessageHandler(provider));
+        }
+
+        public MAuthClient(ITraceProvider provider, HttpMessageHandler handler)
+        {
+            _handler = new MAuthMessageHandler(new ZipkinMessageHandler(provider, handler));
+        }
+
+        public async Task<string> MakeRequest(string uri)
+        {
+            var client = new HttpClient(_handler);
```

Comment:
We can do
```
using (var client = new HttpClient(_handler, false))
{
}
```
I have not tried that though but according to the following article, we can avoid to kick in dispose for the handler --> https://msdn.microsoft.com/en-us/library/hh875108(v=vs.118).aspx